server/src/controllers/reportController.ts

```diff
@@ -0,0 +1,65 @@
+import ReportUploadRequestHandler from '../useCases/report/reportSave/handleSaveRequest';
+import { ReportGetProductRequestHandler } from '../useCases/report/productCompatibility/handleGetProductRequest';
+import ProductCountRequestHandler from '../useCases/report/productCount/handleGetProductCountRequest';
+import ProductDetailsRequestHandler from '../useCases/report/productTestCases/handleGetProductDetailsRequest';
+import { default500Error } from './controllerUtils';
+import ReportGetBuildingBlocksRequestHandler from '../useCases/report/buildingBlocks/handleGetBuildingBlocksRequest';
+import { Request, Response } from 'express';
+import 'multer';
+
+export interface ReportController {
+  saveReport(req: Request, res: Response): void;
+  getProductCompatibility(req: Request, res: Response): void;
+  getProductsCount(req: Request, res: Response): void;
+  getProductDetails(req: Request, res: Response): void;
+  getBuildingBlocks(req: Request, res: Response): void;
+}
+
+const reportController = (reportDbRepository: any, reportDbRepositoryImpl: any): ReportController => {
```

Comment:
I don't think any is a good type for the db repositories. we should use DB Repository interface type in both

Reply:
okay I will add new interface for it